apps/userdashboard/assets/js/a4_candy_userdashboard/ModerationNotification.jsx

```diff
@@ -23,35 +23,44 @@ export default class ModerationNotification extends Component {
     )
   }
 
-  toggleIsPending () {
-    const data = { is_pending: !this.state.isPending }
+  safeFetch = (payload) => {
     const headers = {
       Accept: 'application/json',
       'Content-Type': 'application/json',
       'X-CSRFToken': Cookies.get('csrftoken')
     }
-
-    fetch(this.props.apiUrl, {
+    return fetch(this.props.apiUrl, {
       method: 'PATCH',
       headers: headers,
-      body: JSON.stringify(data)
+      body: JSON.stringify(payload)
     })
       .then(res => res.json())
-      .then(json => {
-        this.props.onChangePending(json.is_pending)
-        this.setState({
-          isPending: json.is_pending
-        })
-      }).catch((err) => {
-        console.log(err)
-      })
+      .then(response => [response, undefined])
+      .catch(error => Promise.resolve([undefined, error]))
   }
 
-  toggleIsBlocked () {
-    const newIsBlocked = !this.state.isBlocked
-    this.setState({
-      isBlocked: newIsBlocked
-    })
+  async toggleIsPending () {
+    const [response, error] =
+      await this.safeFetch({ is_pending: !this.state.isPending })
+    const alertMessage = response && response.is_pending
+      ? 'unarchived'
```

Comment:
I think this needs to be translatable as well?

Reply:
that is true. i will think about a different solution, because using these translated words to identify is a bit too error prone.

Reply:
@Rineee okay i shifted the translation of the word to the end as well, so i can keep these words as identifiers.